File: src/content/docs/topics/cs/coding-problems/arrays-and-hashing/217-contains-duplicate-approach2.go
package main

import (
	"fmt"
	"sort"
)

func assert(condition bool, msgs ...string) {
	if !condition {
		msg := "assertion failed"
		if len(msgs) > 0 {
			msg = msgs[0]
		}
		panic(msg)
	}
}

func containsDuplicate(nums []int) bool {
	sorted := make([]int, len(nums))
	copy(sorted, nums)
	sort.Ints(sorted)
	for i := 1; i < len(sorted); i++ {
		if sorted[i] == sorted[i-1] {
			return true
		}
	}
	return false
}

func runTests() {
	assert(containsDuplicate([]int{1, 2, 3, 1}) == true)
	assert(containsDuplicate([]int{1, 2, 3, 4}) == false)
	assert(containsDuplicate([]int{1, 1, 1, 3, 3, 4, 3, 2, 4, 2}) == true)
	assert(containsDuplicate([]int{}) == false)
	assert(containsDuplicate([]int{5}) == false)
	assert(containsDuplicate([]int{5, 5}) == true)
	fmt.Println("all tests pass")
}

func main() {
	runTests()
}

File: src/content/docs/topics/cs/coding-problems/arrays-and-hashing/217-contains-duplicate-approach2.ts
function assert(condition: boolean, msg: string = ''): void {
    if (!condition) throw new Error(msg || 'Assertion failed');
}

function containsDuplicate(nums: number[]): boolean {
    const sorted = [...nums].sort((a, b) => a - b);
    for (let i = 1; i < sorted.length; i++) {
        if (sorted[i] === sorted[i - 1]) return true;
    }
    return false;
}

assert(containsDuplicate([1, 2, 3, 1]) === true);
assert(containsDuplicate([1, 2, 3, 4]) === false);
assert(containsDuplicate([1, 1, 1, 3, 3, 4, 3, 2, 4, 2]) === true);
assert(containsDuplicate([]) === false);
assert(containsDuplicate([5]) === false);
assert(containsDuplicate([5, 5]) === true);
console.log("all tests pass");
